src/main/java/net/jodah/typetools/TypeResolver.java

```diff
@@ -47,15 +51,43 @@
   private static volatile boolean CACHE_ENABLED = true;
   private static boolean RESOLVES_LAMBDAS;
   private static Method GET_CONSTANT_POOL;
+  private static Method GET_CONSTANT_POOL_SIZE;
+  private static Method GET_CONSTANT_POOL_METHOD_AT;
   private static final Map<String, Method> OBJECT_METHODS = new HashMap<String, Method>();
   private static final Map<Class<?>, Class<?>> PRIMITIVE_WRAPPERS;
   private static final Double JAVA_VERSION;
 
   static {
     JAVA_VERSION = Double.parseDouble(System.getProperty("java.specification.version", "0"));
+
     try {
+      Unsafe unsafe = AccessController.doPrivileged(new PrivilegedExceptionAction<Unsafe>() {
+        @Override
+        public Unsafe run() throws Exception {
+          final Field f = Unsafe.class.getDeclaredField("theUnsafe");
+          f.setAccessible(true);
+
+          return (Unsafe) f.get(null);
+        }
+      });
+
       GET_CONSTANT_POOL = Class.class.getDeclaredMethod("getConstantPool");
-      GET_CONSTANT_POOL.setAccessible(true);
+      String constantPoolName = JAVA_VERSION < 9 ? "sun.reflect.ConstantPool" : "jdk.internal.reflect.ConstantPool";
+      Class<?> constantPoolClass = Class.forName(constantPoolName);
+      GET_CONSTANT_POOL_SIZE = constantPoolClass.getDeclaredMethod("getSize");
+      GET_CONSTANT_POOL_METHOD_AT = constantPoolClass.getDeclaredMethod("getMethodAt", int.class);
+
+      // setting the methods as accessible
+      Field overrideField = AccessibleObject.class.getDeclaredField("override");
+      long overrideFieldOffset = unsafe.objectFieldOffset(overrideField);
+      unsafe.putBoolean(GET_CONSTANT_POOL, overrideFieldOffset, true);
+      unsafe.putBoolean(GET_CONSTANT_POOL_SIZE, overrideFieldOffset, true);
+      unsafe.putBoolean(GET_CONSTANT_POOL_METHOD_AT, overrideFieldOffset, true);
+
+      // additional checks - make sure we get a result when invoking the Class::getConstantPool and
+      // ConstantPool::getSize on a class
+      Object constantPool = GET_CONSTANT_POOL.invoke(Object.class);
+      GET_CONSTANT_POOL_SIZE.invoke(constantPool);
```

Comment:
I picked up that approach to protect against having the errors thrown at runtime.
The check is done before setting the `RESOLVES_LAMBDAS` flag to true.
I think it is better to disable lambdas instead of getting errors at runtime if something does not work as expected.
This issue might happen if the java 9 module security mecanism will change in the future such that it cannot be by-passed.